spec/features/idv/doc_auth/document_capture_spec.rb

```diff
@@ -298,9 +308,353 @@
         end
 
         context 'selfie with error is uploaded' do
+          before do
+            allow(IdentityConfig.store).to receive(:doc_auth_max_attempts).and_return(99)
+
+            allow_any_instance_of(FederatedProtocols::Oidc).
+              to receive(:biometric_comparison_required?).
+              and_return(true)
+            perform_in_browser(:mobile) do
+              visit_idp_from_oidc_sp_with_ial2(biometric_comparison_required: true)
+              sign_in_and_2fa_user(@user)
+              complete_doc_auth_steps_before_document_capture_step
+            end
+          end
+
+          it 'shows the correct error message for the given error' do
+            # when the only error is a doc auth error
+
+            perform_in_browser(:mobile) do
+              attach_images(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_doc_auth_fail_selfie_pass.yml'
+                ),
+              )
+              attach_selfie(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_doc_auth_fail_selfie_pass.yml'
+                ),
+              )
+
+              submit_images
+
+              h1_error_message = strip_tags(t('errors.doc_auth.rate_limited_heading'))
+              expect(page).to have_content(h1_error_message)
+
+              body_error_message = strip_tags(t('doc_auth.errors.dpi.top_msg'))
+              expect(page).to have_content(body_error_message)
+
+              click_try_again
+              expect(page).to have_current_path(idv_document_capture_path)
+
+              inline_error_message = strip_tags(t('doc_auth.errors.dpi.failed_short'))
+              expect(page).to have_content(inline_error_message)
+
+              expect(page).to have_current_path(idv_document_capture_url)
+
+              # when doc auth result passes but liveness fails
+
+              attach_images(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_no_liveness.yml'
+                ),
+              )
+              attach_selfie(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_no_liveness.yml'
+                ),
+              )
+
+              submit_images
+
+              h1_error_message = strip_tags(
+                t('errors.doc_auth.selfie_not_live_or_poor_quality_heading'),
+              )
+              expect(page).to have_content(h1_error_message)
+
+              body_error_message = strip_tags(t('doc_auth.errors.alerts.selfie_not_live'))
+              expect(page).to have_content(body_error_message)
+
+              click_try_again
```

Comment:
I think we need to check the body and h1 that appear on the page here? It should be `Check your images and try again`, `Try taking a photo of yourself again. Make sure your whole face is clear and visible in the photo. Review more tips for taking a clear photo of yourself`, and `1. Driver’s license or state ID card` but maybe we're checking this elsewhere?

Reply:
I see what you mean. I think this is a good idea but I do think most of that is inline error messaging and the same thing [for this comment](https://github.com/18F/identity-idp/pull/10450/files#r1572758817) would apply here (update to be what we know is wrong or add a comment to come back to).